tests/RobotFramework/tests/mqtt/custom_sub_topics_tedge-mapper-c8y.robot

```diff
@@ -0,0 +1,29 @@
+*** Settings ***
+Documentation       Purpose of this test is to verify that tedge-mapper-c8y subscribes the topics that are configured as c8y.topics
+
+Resource            ../../resources/common.resource
+Library             ThinEdgeIO
+
+Suite Setup         Custom Setup
+Suite Teardown      Get Logs
+
+Test Tags           theme:mqtt    theme:c8y
+
+
+*** Test Cases ***
+Publish events to subscribed topic
+    Execute Command    mosquitto_pub -t tedge/events/event-type -m '{"text": "Event"}'
+    Should Have MQTT Messages    c8y/s/us
```

Comment:
You have to be more specific with your comparison as there might be other messages that are published to `c8y/s/us` which will cause the test to pass.

The `Should Have MQTT Messages` supports a message pattern named argument, `message_pattern`:

```sh
Should Have MQTT Messages    c8y/s/us    message_pattern=.*Event.*
```

Or something to that effect (not 100% sure on what the expected payload is in this case, I would have expected it to be published on the json via mqtt topic `c8y/event/events`...but I can't remember the exact implementation for events).

Reply:
Actually, event is converted to SmartREST like below.

```
[c8y/s/us] 400,event-type,"Event",2023-06-28T17:30:21.529621713Z
```

Reply:
Though I’m sure there are other messages being sent to the same topic, so we have to be sure the assertion is specific enough.